receiver/ocmetrics/metrics_receiver.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocmetrics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+
+	"google.golang.org/grpc"
+
+	agentmetricspb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/metrics/v1"
+	"github.com/census-instrumentation/opencensus-service/internal"
+	"github.com/census-instrumentation/opencensus-service/metricsink"
+	"github.com/census-instrumentation/opencensus-service/receiver"
+)
+
+// NewMetricsReceiver will create a handle that runs an OCMetricsReceiver at the provided port.
+func NewMetricsReceiver(port int, opts ...Option) (receiver.MetricsReceiver, error) {
+	return &ocReceiverHandler{srvPort: port, receiverOptions: opts}, nil
+}
+
+// NewMetricsReceiverOnDefaultPort will create a handle that runs an OCMetricsReceiver at the default port.
+func NewMetricsReceiverOnDefaultPort(opts ...Option) (receiver.MetricsReceiver, error) {
+	return &ocReceiverHandler{srvPort: defaultOCMetricsReceiverPort, receiverOptions: opts}, nil
+}
+
+type ocReceiverHandler struct {
+	mu      sync.RWMutex
```

Comment:
nit: since only Lock() and Unlock() are used with mu use a `sync.Mutex` instead.

Reply:
Done.